src/application.rs

```diff
@@ -120,35 +120,35 @@ impl From<String> for OAuthToken {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone, PartialEq)]
 pub enum Application {
     AppId(AppId, Option<AppKey>),
     UserKey(UserKey),
     OAuthToken(OAuthToken)
 }
 
 // These trait impls build an Application variant out of its required types
-impl Into<Application> for AppId {
-    fn into(self) -> Application {
-        Application::AppId(self, None)
+impl From<AppId> for Application {
+    fn from(a: AppId) -> Self {
```

Comment:
I think that `-> Application` makes the code more readable. Any reason to use `Self` here?

Reply:
Yes, many.

It's DRY, changes less code when refactoring the type name, comes with benefits (ie. think how methods would look with `Self = Iterator<Result<Arc<MyType<T>>, E>>`), it is becoming idiomatic Rust after being stabilised and implemented in more places of the language (ie. `Self(1,2,3)` for `(u8, u8, u8)`).

It is also the only thing trait methods use for constructing values of the type implementing the trait (so common for constructors to have a `-> Self` signature), and also all the `self`, `&self`, `&mut self` in parameters are sugar for `self: [&]Self`, which then would be sugar for `self: [&]Application` (but not `application: [&]Application`).

Well, and it reads better to me! :)